third_party/move/move-vm/runtime/src/runtime.rs

```diff
@@ -121,8 +134,11 @@ impl VMRuntime {
         }
 
         // Perform bytecode and loading verification. Modules must be sorted in topological order.
-        self.loader
-            .verify_module_bundle_for_publication(&compiled_modules, data_store)?;
+        self.loader.verify_module_bundle_for_publication(
+            &compiled_modules,
+            data_store,
```

Comment:
Do you know why do we have `data_store` here and in `load_script `/`load_function`? Isn't code decoupled from data?

Reply:
So the logic is that `data_store` gives you the serialized module data or move value data. The `module_store` is only for storing the resolved modules. The naming here is a bit confusing tho...

Reply:
It's for staging the outputs, right? we should never have to cache the module bytes (that we obtain via get_module_bytes, so it's decoupled from data/resource storage all the way on the read side).

Does the current method (publish_module_bundle) just check whether module bundle can be published, or does it actually put anything new in data_store? It seems it passes mut data store only because it may load old versions of modules? In that case publish_module_bundle name can be quite misleading imho.

I think at some point we can introduce a feature flag, and put the new CompiledModules in the "local" module cache, which should also make the init_module behavior a lot more intuitive.